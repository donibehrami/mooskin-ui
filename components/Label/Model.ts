import { IBoxComponentProps } from '../Box/model';

export interface ILabelComponentProps extends IBoxComponentProps {
    /** Label children */
    children: string;

    /** Make Label disabled */
    disabled?: boolean;

    /** callback that is called when the Label is clicked */
    onClick?: (e: React.MouseEvent<HTMLElement>) => void;
}
