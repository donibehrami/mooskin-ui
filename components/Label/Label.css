@import url('https://cdn.moostaging.com/assets/fonts/google-web-fonts/fonts.css');

.label{
    font-family: 'Convergence', 'Alegreya Sans', sans-serif;
    font-size: 0.875rem;
}