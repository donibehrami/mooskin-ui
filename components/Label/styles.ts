import styled from 'styled-components';

// Models
import { ILabelComponentProps } from './model';

// Components
import Box from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledLabel = styled(Box)<ILabelComponentProps>`
    font-family: Montserrat;
    font-size: 14px;
    font-weight: 500;
    color: ${(props) => props.disabled ? props.theme.disabledfont || variables.disabledfont : props.theme.label || variables.label};
    display: flex;
    align-self: center;
    cursor: ${(props) => (!props.disabled ? (props.onClick ? 'pointer' : 'unset') : 'not-allowed')};
    min-width: ${(props) => props.width || 150}px;
`;
