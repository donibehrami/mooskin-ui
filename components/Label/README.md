# MooSkin UI - Label Component

The MooSkin Label Component, a simple labeling component with pre-defined styling and attributes.

___

### Usage

To start using the Label Component first you have to Import it

```
Import {Label} from 'mooskin';
```
or modular import
```
// JS
import Label from 'mooskin/lib/Label';

// CSS
import 'mooskin/lib/Label/style.css';
```

And then you can simply start using it by typing

```
<Label attribute1="atr" attribute2="asd" >Label text</Label>
```

The component it will accept given attributes and render differently based on the given attributes

### Examples


Making it disabled, in this case the proper disabled class will be loaded (ex. disabling cursor)


Give it custom styles

```
<Label style={yourStyle} >Label content</Label>
```

or abbreviate numerical values

```
<Label abbreviate>13400</Label>
```

this returns a label with a value of 13.4K. Works with higher values aswell.

<div class="playground-doc">

## Supported attributes

* `id` - id of the element
* `className` - css class
* `style` - button style
* `abbreviate` - abbreviates large numerical values to K, M, B, and T.

</div>

Allthough these attributes are supported, all of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)