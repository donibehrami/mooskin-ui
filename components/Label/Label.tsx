import * as React from 'react';
import variables from '../_utils/globals/variables';

import { ILabelComponentProps } from './model';
import { StyledLabel } from './styles';

export const Label: React.FC<ILabelComponentProps> = (props) => {
    const onClick = (e: React.MouseEvent<HTMLElement>) => {
        !props.disabled && props.onClick && props.onClick(e);
    };

    return <StyledLabel {...props} onClick={onClick} />;
};

Label.defaultProps = {
    className: '',
    primaryColor: variables.label,
    style: {},
};

Label.displayName = 'Label';

export default Label;
