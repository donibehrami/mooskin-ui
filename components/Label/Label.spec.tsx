import * as React from 'react';
import Label from './Label';

import { shallow } from 'enzyme';

describe('Button', () => {

    test('renders correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Label
                className="myClass"
                style={{color: 'blue'}}
                id={'label'}
                abbreviate
            >
                Mooskin
            </Label>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders simple text with label styles', () => {

        const component = shallow(<Label>Mooskin</Label>);

        expect(component.find('div').text()).toBe('Mooskin');
    });

    test('renders simple numbers text', () => {

        const component = shallow(<Label>12345</Label>);

        expect(component.find('div').text()).toBe('12345');
    });

    test('abbreviates number values if type prop of number is passed (thousands)', () => {

        const component = shallow(<Label type="number">13400</Label>);

        expect(component.find('div').text()).toBe('13.4K');
    });

    test('abbreviates number values if type prop of number prop is passed (millions)', () => {

        const component = shallow(<Label type="number">3235942</Label>);

        expect(component.find('div').text()).toBe('3.2M');
    });

    test('abbreviates number values if type prop of number prop is passed (billions)', () => {

        const component = shallow(<Label type="number">6345153975</Label>);

        expect(component.find('div').text()).toBe('6.3B');
    });

    test('abbreviates number values if type prop of number prop is passed (trillion - enthusiast mode)', () => {

        const component = shallow(<Label type="number">8675345876235</Label>);

        expect(component.find('div').text()).toBe('8.6T');
    });

    test('abbreviates number values and rounds them if type prop of round_number is passed (thousands)', () => {

        const component = shallow(<Label type="round_number">13400</Label>);

        expect(component.find('div').text()).toBe('13K');
    });

    test('abbreviates number values and rounds them if type prop of round_number is passed (millions)', () => {

        const component = shallow(<Label type="round_number">3235942</Label>);

        expect(component.find('div').text()).toBe('3M');
    });

    test('abbreviates number values and rounds them if type prop of round_number is passed (billions)', () => {

        const component = shallow(<Label type="round_number">6545153975</Label>);

        expect(component.find('div').text()).toBe('7B');
    });

    test('abbreviates number values and rounds them if type prop of round_number is passed (trillion)', () => {

        const component = shallow(<Label type="round_number">8675345876235</Label>);

        expect(component.find('div').text()).toBe('9T');
    });

});
