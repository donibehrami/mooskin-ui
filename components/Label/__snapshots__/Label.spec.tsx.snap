// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<Label
  className=""
  style={Object {}}
>
  <Component
    className=""
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "common": "#ffffff",
          "disabledButton": "#e2e2e2",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    style={Object {}}
  >
    <StyledLabel
      boxAs="label"
      className=""
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "common": "#ffffff",
            "disabledButton": "#e2e2e2",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      style={Object {}}
    >
      <Box
        boxAs="label"
        className="sc-gtssRu eKLwVr"
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "common": "#ffffff",
              "disabledButton": "#e2e2e2",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        style={Object {}}
      >
        <Component
          boxAs="label"
          className="sc-gtssRu eKLwVr"
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "common": "#ffffff",
                "disabledButton": "#e2e2e2",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          style={Object {}}
        >
          <StyledBox
            as="label"
            boxAs="label"
            className="sc-gtssRu eKLwVr"
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "common": "#ffffff",
                  "disabledButton": "#e2e2e2",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            style={Object {}}
          >
            <label
              className="sc-bdnylx sJSbd sc-gtssRu eKLwVr"
              style={Object {}}
            >
              Label here!
            </label>
          </StyledBox>
        </Component>
      </Box>
    </StyledLabel>
  </Component>
</Label>
`;
